packages/react-output-target/react-component-lib/createComponent.tsx

```diff
@@ -54,14 +55,23 @@ export const createReactComponent = <
     render() {
       const { children, forwardedRef, style, className, ref, ...cProps } = this.props;
 
-      let propsToPass = Object.keys(cProps).reduce((acc, name) => {
+      let propsToPass = Object.keys(cProps).reduce((acc: any, name) => {
+        const value = (cProps as any)[name];
+
         if (name.indexOf('on') === 0 && name[2] === name[2].toUpperCase()) {
           const eventName = name.substring(2).toLowerCase();
+
           if (typeof document !== 'undefined' && isCoveredByReact(eventName, document)) {
-            (acc as any)[name] = (cProps as any)[name];
+            acc[name] = value;
           }
         } else {
-          (acc as any)[name] = (cProps as any)[name];
+          // we should only render strings, booleans, and numbers as attrs in html.
+          // objects, functions, arrays etc get synced via properties on mount.
+          const type = typeof value;
+
+          if (type === 'string' || type === 'boolean' || type === 'number') {
```

Comment:
should we be checking if `reflect: true` for each prop?

Reply:
I don't think so? Reflect controls whether internal updates to a prop causes a corresponding update in the attr,  right? It doesn't affect how attributes get written from the outside

Reply:
i know, but right now if a prop is not reflecting, even if u set it in React, it will update the prop without needing to add an attribute, so why change that?

Reply:
The point is to get as much information into static html as possible, to improve SSR. Any time something is passed as a property you need to wait until after first render, so you get layout shift and flashes of styles as things update

Reply:
i see, then ignore this one